### ISSUE TITLE
[Bug]: Heap corruption due to virtual function call from destructor of AccessibilityHandler 

### ISSUE THREAD

**Issue by NickolayGerasimenko** [juce-framework/JUCE#1156]
### Detailed steps on how to reproduce the bug

Inherit from `ListBoxModel` and override virtual function `getNumRows`
During destruction of `SectionedListBox` virtual function getNumRows will be called **for partially destroyed
object of class SectionedListBox** from inside destructor of `AccessibilityHandler`.

Moreover, depending on implementation of `SectionedListBox::getNumRows` potentially this can lead to pure-virtual function call [error.](<img width="697" alt="Screenshot 2022-12-01 at 22 00 25" src="https://user-images.githubusercontent.com/807345/205158537-0b8aecc1-1257-4f17-8306-ef20e266fec7.png">)

The implementation of AccessibilityHandler violates C.82 Rule of C++ Core Guidline:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c82-dont-call-virtual-functions-in-constructors-and-destructors

```
class SectionedListBox: public juce::Component,
    private juce::ListBoxModel
{
public:;
    // juce::ListBoxModel interfaces
    int getNumRows() override;
```

### workaround:

disable `AccessibilityHandler` for `SectionedListBox`

```
SectionedListBox::SectionedListBox()
{
    setAccessible(false);
}
```

### stacktrace

<img width="697" alt="Screenshot 2022-12-01 at 22 00 25" src="https://user-images.githubusercontent.com/807345/205158724-afa59ee7-7129-4b8b-9234-f644332a835d.png">

### What is the expected behaviour?

AccessibilityHandler should be properly destroyed.

### Operating systems

Windows, macOS

### What versions of the operating systems?

OSX 12.6.1

### Architectures

x86_64

### Testing on the `develop` branch

The bug is present on the `develop` branch

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by tpoole** [comment]
Could you please provide a minimal example that displays the issue? From the stack trace it looks like there's a lot more going on than just `getNumRows` overridden. Can you reproduce this problem in the DemoRunner? Are you using a custom component for a row?

**Comment by NickolayGerasimenko** [comment]
@tpoole Custom component for `ListBoxModel`
```
class SectionedListBox: public juce::Component,
    private juce::ListBoxModel
{
```

**Comment by tpoole** [comment]
What else is happening in that component? There's a call to `~LabelAccessibilityHandler` that appears to be the problematic bit, but it's not clear where that's coming from.
